src/test/java/TypewiseAlert/TypewiseAlertTest.java

```diff
@@ -4,12 +4,31 @@
 
 import org.junit.Test;
 
-public class TypewiseAlertTest 
-{
-    @Test
-    public void infersBreachAsPerLimits()
-    {
-      assertTrue(TypewiseAlert.inferBreach(12, 20, 30) ==
-        TypewiseAlert.BreachType.TOO_LOW);
-    }
+
+public class TypewiseAlertTest {
+	@Test
+	public void infersBreachAsPerLimits() {
+		assertTrue(TypewiseAlert.inferBreach(12, 20, 30) == BreachType.TOO_LOW);
+		assertTrue(TypewiseAlert.inferBreach(40, 30, 40) == BreachType.NORMAL);
+		assertTrue(TypewiseAlert.inferBreach(55, 45, 50) == BreachType.TOO_HIGH);
+	}
+	
+	@Test
+	public void verifyBreachType() {
+		assertTrue(TypewiseAlert.classifyTemperatureBreach(CoolingType.PASSIVE_COOLING, -10) == BreachType.TOO_LOW);
```

Comment:
tests to assert upper and lower limits for each cooling type is needed , otherwise many input combinations are needed for testing classifyTemperatureBreach() method